src/sessions/video-session-handler.ts

```diff
@@ -309,81 +311,87 @@ export default class VideoSessionHandler extends BaseSessionHandler {
         }
       }
 
-      session.emit('incomingMedia');
+      session.emit(SessionEvents.incomingMedia);
     };
 
-    if (session.tracks.length) {
-      handleIncomingTracks(session, session.tracks);
+    const tracks = session.pc.getReceivers()
+      .filter(receiver => receiver.track)
+      .map(receiver => receiver.track);
+
+    if (tracks.length) {
+      handleIncomingTracks(session, tracks);
     } else {
-      session.on('peerTrackAdded', (session: IJingleSession, track: MediaStreamTrack) => {
+      session.on(MediaSessionEvents.peerTrackAdded, (track: MediaStreamTrack) => {
         handleIncomingTracks(session, track);
       });
     }
 
     await super.acceptSession(session, params);
-    this.setInitialMuteStates(session);
+    await this.setInitialMuteStates(session);
 
     logDeviceChange(this.sdk, session, 'sessionStarted');
   }
 
-  setInitialMuteStates (session: IJingleSession): void {
+  async setInitialMuteStates (session: IExtendedMediaSession): Promise<void> {
     const userId = this.sdk._personDetails.id;
 
+    let videoMute = Promise.resolve();
+    let audioMute = Promise.resolve();
+
     const videoSender = session.pc.getSenders().find((sender) => sender.track && sender.track.kind === 'video');
     if (!videoSender || !videoSender.track.enabled) {
       session.videoMuted = true;
       this.log(LogLevels.info, 'Sending initial video mute', { conversationId: session.conversationId, sessionId: session.id });
-      session.mute(userId, 'video');
-    } else {
-      session.videoMuted = false;
+      videoMute = session.mute(userId as any, 'video');
     }
 
     const audioSender = session.pc.getSenders().find((sender) => sender.track && sender.track.kind === 'audio');
     if (!audioSender || !audioSender.track.enabled) {
       session.audioMuted = true;
       this.log(LogLevels.info, 'Sending initial audio mute', { conversationId: session.conversationId, sessionId: session.id });
-      session.mute(userId, 'audio');
-    } else {
-      session.audioMuted = false;
```

Comment:
Is `audioMuted` and `videoMuted` still on the `session` object? The reason this `else` statement is here is for logging purposes when calling `logDeviceChange(this.sdk, session, 'sessionStarted');` on line 332. But I guess that function could be updated to convert `undefined` to `false` so it actually gets logged correctly in sumo.

Reply:
they are in the sdk, but we have to track/change it manually, which should already be happening.